evc-sumo/src/evc_connector.py

```diff
@@ -88,23 +84,51 @@ def get_traffic_light_status_from_EVC(self, controller_io, phases):
         type: dict
 
         Returns:
-        - string: A string representing the current state of the traffic light. The possible characters are "r" for red, "y" for yellow, and "g" for green.
+        - string: A string representing the current state of the traffic light. 
+                  The possible characters are "r" for red, "y" for yellow, and "g" for green and None for exception
         """
 
-        ## get number of characters in state string for SUMO
-        state_num = 0
-        for phase in phases:
-            state_num = state_num + len(phase["sumoTlStateIndex"])
-        ## init sumo tl state string with all red states
-        state_string = ['r'] * state_num
+        ## get number of characters in state string for SUMO and init sumo tl state string with all red
+        state_string = ['r'] * sum(len(phase['sumoTlStateIndex']) for phase in phases)
 
         for phase in phases:
             state_string = [self.COB_to_traffic_light_status(controller_io, phase['evcPhaseId']) if i in phase['sumoTlStateIndex'] else x for i,x in enumerate(state_string)]
-        return ''.join(state_string)
+        if None in state_string:
```

Comment:
I don't think None can be an element of a String right? A string variable could, itself be None, but I don't think you could have "rrrrryyNonegggg" or something. Or, after a quick google search you can but it would be that 
```
  a = "foo"
  b = None
  c = "bar
   
  out = a + b + c
  
```

out would be the string "fooNonebar" where None is simply text, not the actual None literal.